python/lsst/ts/ATDomeTrajectory/dome_trajectory.py

```diff
@@ -61,7 +65,10 @@ class ATDomeTrajectory(salobj.BaseCsc):
     * 1: simulation mode: start a mock ATDome controller and talk to it
       using SAL.
     """
-    def __init__(self, initial_state=salobj.base_csc.State.STANDBY, initial_simulation_mode=0):
+    def __init__(self, config_dir=None, initial_state=salobj.base_csc.State.STANDBY,
+                 initial_simulation_mode=0):
+        schema_path = pathlib.Path(__file__).parents[4].joinpath("schema", "ATDomeTrajectory.yaml")
```

Comment:
I imagine you though about this but, is there a way to make this discovery in the superclass? 

I imagine people will have to do this every time so I'd hope it should be done by `ConfigurableCsc`.

Reply:
I agree. The problem is that I don't know how to reliably find the package root directory. It may not always be "4 levels up from the CSC definition" because the CSC might be defined deeper down.

If we were defining the eups <packagename>_DIR environment variables for deployment then it would be trivial. But I don't want to assume that for all CSC packages. That said, the code that finds the default config directory does need one of those already (for, e.g. `$TS_CONFIG_OCS_DIR`), so I suppose it would not kill us to define an additional env variable for the CSC package as well.

What are your thoughts?

Reply:
What if we use a relative path instead? How bad would it be if the `schema/` folder lived in the same level as the csc code, instead of the package upper level?

Reply:
Personally that makes me very uncomfortable, as it puts the schema right in the thick of python library code.

Reply:
I thought about this some more and I will make schema_path default to None, and if None then ConfigurableCsc computes a path based on the standard layout. That should work for most CSCs, and violators can provide an explicit path.

Reply:
I tried it and found that it's not easy to get the __file__ of the subclass. One has to play tricks with importlib (e.g. https://stackoverflow.com/a/54142935) and I'm not sure they are robust. I'm going back to the original code.